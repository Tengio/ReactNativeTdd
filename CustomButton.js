import React, { Component } from 'react';
import { View, Text } from 'react-native';

export default class CustomButton extends React.Component {
    render() {
        return <View>
                    <Text>{this.props.text}</Text>
                </View>
    }
}