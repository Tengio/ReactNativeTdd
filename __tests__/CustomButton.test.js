import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../CustomButton';

describe('CustomButton', () => {
    
    it('should render with undefined text', () => {
        const tree = renderer.create(
            <CustomButton />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should render text', () => {
        const tree = renderer.create(
            <CustomButton text="Submit"/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})