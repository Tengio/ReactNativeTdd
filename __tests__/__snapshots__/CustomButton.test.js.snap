// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should render text 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Submit
  </Text>
</View>
`;

exports[`CustomButton should render with undefined text 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
